tcp/client.go

```diff
@@ -0,0 +1,86 @@
+package tcp
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/influxdb/influxdb/data"
+)
+
+type Client struct {
+	conn net.Conn
+}
+
+func NewClient() *Client {
+	return &Client{}
+}
+
+func (c *Client) Dial(addr string) error {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return err
+	}
+
+	c.conn = conn
+	return nil
+}
+
+func (c *Client) WriteShardRequest(shardID uint64, points []data.Point) error {
```

Comment:
Could probably make this `WriteShard` so it implements the `ShardWriter` interface.